timemachine/md/minimizer.py

```diff
@@ -172,15 +183,94 @@ def minimize_host_4d(mols, host_system, host_coords, ff, box, mol_coords=None) -
     final_coords = fire_minimize(x, u_impls, box, 50)
     for impl in u_impls:
         du_dx, _ = impl.execute(final_coords, box)
-        norm = np.linalg.norm(du_dx, axis=-1)
-        if not np.all(norm < MAX_FORCE_NORM):
-            raise MinimizationError(
-                f"Minimization failed to reduce large forces below threshold: |frc| = {norm} > {MAX_FORCE_NORM}"
-            )
+        check_force_norm(du_dx)
 
     return final_coords[:num_host_atoms]
 
 
+def make_gpu_impl(potentials):
+    """return bound impl of a SummedPotential constructed from potentials"""
+
+    params = [p.params for p in potentials]
+    flat_params = np.concatenate([param.reshape(-1) for param in params])
+
+    generic_potentials = [generic.from_gpu(p) for p in potentials]
+    summed_potential = generic.SummedPotential(generic_potentials, params)
+
+    return summed_potential.to_gpu().bind(flat_params).bound_impl(np.float32)
+
+
+def make_host_du_dx_fxn(mols, host_system, host_coords, ff, box, mol_coords=None):
+    """construct function to compute du_dx w.r.t. host coords, given fixed mols and box"""
+
+    # openmm host_system -> timemachine host_bps
+    host_bps, _ = openmm_deserializer.deserialize_system(host_system, cutoff=1.2)
+
+    # construct appropriate topology from (mols, ff)
+    if len(mols) == 1:
+        top = topology.BaseTopology(mols[0], ff)
+    elif len(mols) == 2:
+        top = topology.DualTopologyMinimization(mols[0], mols[1], ff)
+    else:
+        raise ValueError("mols must be length 1 or 2")
+
+    hgt = topology.HostGuestTopology(host_bps, top)
+
+    # bound impls of potentials @ lam=0 (fully coupled) endstate
+    params_potential_pairs = parameterize_system(hgt, ff, 0.0)
+    bound_potentials = [potential.bind(params) for params, potential in params_potential_pairs]
+    gpu_impl = make_gpu_impl(bound_potentials)
+
+    # read conformers from mol_coords if given, or each mol's conf0 otherwise
+    conf_list = [np.array(host_coords)]
+
+    if mol_coords is not None:
+        for mc in mol_coords:
+            conf_list.append(mc)
+    else:
+        for mol in mols:
+            conf_list.append(get_romol_conf(mol))
+
+    combined_coords = np.concatenate(conf_list)
+
+    # wrap gpu_impl, partially applying box, mol coords
+    num_host_atoms = host_coords.shape[0]
+    assert box.shape == (3, 3)
+
+    def du_dx_host_fxn(x_host):
+        x = np.array(combined_coords)
+        x[:num_host_atoms] = x_host
+
+        du_dx, _ = gpu_impl.execute(x, box)
+        du_dx_host = du_dx[:num_host_atoms]
+        return du_dx_host
+
+    return du_dx_host_fxn
+
+
+def equilibrate_host_barker(
+    mols, host_system, host_coords, ff, box, mol_coords=None, temperature=300.0, proposal_stddev=0.0001, n_steps=1000
```

Comment:
```suggestion
    mols, host_system, host_coords, ff, box, mol_coords=None, temperature=constants.DEFAULT_TEMP, proposal_stddev=0.0001, n_steps=1000
```

Reply:
Applied in https://github.com/proteneer/timemachine/pull/978/commits/97196fb589c4b32c54cd2be832f7c1d9bf665058